### ISSUE TITLE
Bug: insert_image hangs forever with a specific image

### ISSUE THREAD

**Issue by rajyan** [jmcnamara/rust_xlsxwriter#117]
### Current behavior

rust_xlsxwriter hangs forever when this image is inserted.
![convert](https://github.com/user-attachments/assets/7605610c-d525-4783-8a18-2adb00a5f5da)

### Expected behavior

The process should finish (at least with an error).

### Sample code to reproduce

```rust
#!/usr/bin/env rust-script

//! cargo
//! [dependencies]
//! rust_xlsxwriter = { version = "0.79.3" }
//! ```
fn main() -> Result<(), rust_xlsxwriter::XlsxError> {
    let mut workbook = rust_xlsxwriter::Workbook::new();
    let worksheet = workbook.add_worksheet();
    let image = rust_xlsxwriter::Image::new("convert.png")?;
    worksheet.insert_image(0, 0, &image)?;

    workbook.save("test.xlsx")?;

    Ok(())
}
```

### Environment

```text
- `rust_xlsxwriter` version: rust_xlsxwriter = { version = "0.79.3" }
- Cargo.toml dependency line for `rust_xlsxwriter`: rust_xlsxwriter = { version = "0.79.3" }
- rustc version: rustc 1.77.0 (aedd173a2 2024-03-17)
```

**Comment by jmcnamara** [comment]
Thanks for the report. I'll look into it

**Comment by rajyan** [comment]
Thank you so much for fixing it so quickly!

confirmed that
```
#!/usr/bin/env rust-script

//! ```cargo
//! [dependencies]
//! rust_xlsxwriter = { git = "https://github.com/jmcnamara/rust_xlsxwriter" }
//! ```
fn main() -> Result<(), rust_xlsxwriter::XlsxError> {
    let mut workbook = rust_xlsxwriter::Workbook::new();
    let worksheet = workbook.add_worksheet();
    let image = rust_xlsxwriter::Image::new("convert.png")?;

    worksheet.insert_image(0, 0, &image)?;

    workbook.save("test.xlsx")?;

    Ok(())
}
```
is working.

**Comment by jmcnamara** [comment]
The fix is now upstream in rust_xlsxwriter v0.79.4. Thanks for the report.
